packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -345,8 +344,34 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
     }
   };
 
+  /**
+   * Adds custom Queries, Mutations, and Subscriptions to the authNonModelConfig map to ensure they are included when adding implicit
+   * aws_iam auth directives.
+   */
+  addCustomOperationFieldsToAuthNonModelConfig = (ctx: TransformerTransformSchemaStepContextProvider): void => {
+    if (!ctx.transformParameters.sandboxModeEnabled && !ctx.synthParameters.enableIamAccess) {
+      return;
+    }
+
+    const hasAwsIamDirective = (field: FieldDefinitionNode): boolean => {
+      return field.directives?.some((dir) => dir.name.value === 'aws_iam');
+    };
```

Comment:
nit: We already have a centralized directive existence checker on Object types [defined in the common package]( https://github.com/aws-amplify/amplify-category-api/blob/9dde73c09dd4f42c666c256c8543a6b94dc4a084/packages/graphql-transformer-common/src/definition.ts#L436) for field types so it is re-usable across transformers

Reply:
Thanks, will address in a followup PR!